diff --git a/Userland/Applications/Maps/main.cpp b/Userland/Applications/Maps/main.cpp
index adf242a2..4807f14c 100644
--- a/Userland/Applications/Maps/main.cpp
+++ b/Userland/Applications/Maps/main.cpp
@@ -7,6 +7,7 @@
 #include "FavoritesPanel.h"
 #include "SearchPanel.h"
 #include "UsersMapWidget.h"
+#include <limits.h>
 #include <LibConfig/Client.h>
 #include <LibCore/System.h>
 #include <LibDesktop/Launcher.h>
diff --git a/Userland/Applications/Presenter/Presentation.cpp b/Userland/Applications/Presenter/Presentation.cpp
index 6e788a45..e4ef39c6 100644
--- a/Userland/Applications/Presenter/Presentation.cpp
+++ b/Userland/Applications/Presenter/Presentation.cpp
@@ -84,7 +84,7 @@ void Presentation::go_to_first_slide()
 ErrorOr<NonnullOwnPtr<Presentation>> Presentation::load_from_file(StringView file_name)
 {
     if (file_name.is_empty())
-        return ENOENT;
+        return Error::from_errno(ENOENT);
     auto file = TRY(Core::File::open_file_or_standard_stream(file_name, Core::File::OpenMode::Read));
     auto contents = TRY(file->read_until_eof());
     auto content_string = StringView { contents };
